modules/vstudio/vslatest.lua

```diff
@@ -0,0 +1,58 @@
+--
+-- vslatest.lua
+-- Extend the existing exporters with support for Visual Studio 2019.
+-- Copyright (c) Jason Perkins and the Premake project
+--
+
+	local p = premake
+	local vstudio = p.vstudio
+	local latest = nil
+
+	function getLatestVisualStudioVersion()
+		-- Get the most recent version installed
+		if latest == nil then
+			local version = nil
+			if os.host() == 'windows' then
+				local ver, err = os.outputof('"C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\vswhere.exe" -latest -property catalog_productLineVersion')
+				if err ~= nil and err == 0 then
+					version = 'vs' .. ver
+				end
+			end
+
+			if version == nil then
+				version = "vs2019"
+				print("No Visual Studio installation found using vswhere.exe. Defaulting to " .. _ACTION .. ".")
+			end
+
+			latest = p.action.get(version)
+		end
+	
+		return latest
+	end
+
+---
+-- Define the Visual Studio Latest export action.
+---
+
+	newaction {
+		-- Metadata for the command line and help system
+		
+		trigger	    = "vs-latest",
+		shortname   = "Visual Studio - Latest",
+		description = "Generate Visual Studio project files for latest installation",
+
+		-- Copy data from the latest vs action
+		
+		targetos         = getLatestVisualStudioVersion().targetos,
```

Comment:
This executes on start-up, not just when this action is used. You can see in the checks, that on non-Windows builds Premake now emits `No Visual Studio installation found using vswhere.exe. Defaulting to test.`

Reply:
Hmm... that's not a behavior we want.   Is there a mechanism that I can use to hold this off until the vs-latest action is invoked? Or would it just be best to only emit the error if `_ACTION == vs-latest`.

Reply:
I don't think so, from my understanding this kind of action is tricky to pull off since you need to execute this kind of thing before the oven/bake process. Your last idea sounds like the best move, but overall, I'm not sold on this new action as I stated in the overall review comment.